src/query/interpreter.cpp

```diff
@@ -1521,73 +1522,156 @@ std::vector<std::vector<TypedValue>> AnalyzeGraphQueryHandler::AnalyzeGraphCreat
+                });
+
+  // Iterate over all label property indexed vertices
+  std::for_each(
+      label_property_indices_info.begin(), label_property_indices_info.end(),
+      [execution_db_accessor, &label_property_counter, &vertex_degree_counter, view](const LPIndex &index_info) {
+        auto vertices = execution_db_accessor->Vertices(view, index_info.first, index_info.second);
+        std::for_each(vertices.begin(), vertices.end(),
+                      [&index_info, &label_property_counter, &vertex_degree_counter, &view](const auto &vertex) {
+                        label_property_counter[index_info][*vertex.GetProperty(view, index_info.second)]++;
+                        vertex_degree_counter[index_info] += *vertex.OutDegree(view) + *vertex.InDegree(view);
+                      });
+      });
+
+  std::vector<std::pair<LPIndex, storage::LabelPropertyIndexStats>> label_property_stats;
+  std::for_each(
+      label_property_counter.begin(), label_property_counter.end(),
+      [execution_db_accessor, &vertex_degree_counter, &label_property_stats](const auto &counter_entry) {
+        const auto &[label_property, values_map] = counter_entry;
+        // Extract info
+        int64_t count_property_value = std::accumulate(
+            values_map.begin(), values_map.end(), 0,
+            [](int64_t prev_value, const auto &prop_value_count) { return prev_value + prop_value_count.second; });
+        // num_distinc_values will never be 0
+        double avg_group_size = static_cast<double>(count_property_value) / static_cast<double>(values_map.size());
+        double chi_squared_stat = std::accumulate(
+            values_map.begin(), values_map.end(), 0.0, [avg_group_size](double prev_result, const auto &value_entry) {
+              return prev_result + utils::ChiSquaredValue(value_entry.second, avg_group_size);
+            });
+
+        double average_degree = count_property_value > 0 ? static_cast<double>(vertex_degree_counter[label_property]) /
+                                                               static_cast<double>(count_property_value)
+                                                         : 0;
+
+        auto index_stats =
+            storage::LabelPropertyIndexStats{.count = count_property_value,
+                                             .distinct_values_count = static_cast<int64_t>(values_map.size()),
+                                             .statistic = chi_squared_stat,
+                                             .avg_group_size = avg_group_size,
+                                             .avg_degree = average_degree};
+        execution_db_accessor->SetIndexStats(label_property.first, label_property.second, index_stats);
+        label_property_stats.push_back(std::make_pair(label_property, index_stats));
+      });
+
+  std::vector<std::vector<TypedValue>> results;
+  results.reserve(label_stats.size() + label_property_stats.size());
```

Comment:
understand and no need to add this now. Shouldn't then this be always a part of memgraph, why would someone even need to run analyze graph to have more optimal query planner. Because one thing is for indices, that is something optional, but why would they have to run query to get better behavior?

Reply:
Yes I would like to add some inherent tracking of index statistics, that's coming in the upcoming PRs so we can eliminate possibly `ANALYZE GRAPH`